business/openshift_oauth.go

```diff
@@ -57,37 +60,59 @@ type OAuthRouteTLSSpec struct {
 const serverPrefix = "https://kubernetes.default.svc/"
 
 func (in *OpenshiftOAuthService) Metadata() (metadata *OAuthMetadata, err error) {
-	var server *OAuthAuthorizationServer
-
-	response, err := request("GET", ".well-known/oauth-authorization-server", nil)
+	redirectURL, err := getKialiRoutePath()
 
 	if err != nil {
 		log.Error(err)
-		message := fmt.Errorf("could not send request to the Openshift OAuth API: %v", err)
+		message := fmt.Errorf("could not get Kiali route for OAuth redirect: %v", err)
 		return nil, message
 	}
 
-	err = json.Unmarshal(response, &server)
+	server, err := getOAuthAuthorizationServer()
+	if err != nil {
+		return nil, err
+	}
 
+	version, err := in.k8s.GetServerVersion()
 	if err != nil {
-		log.Error(err)
-		message := fmt.Errorf("could not parse data from the Openshift API: %v", err)
-		return nil, message
+		return nil, err
 	}
 
-	redirectURL, err := getKialiRoutePath()
+	metadata = &OAuthMetadata{}
+
+	if version.Major == "1" && (strings.HasPrefix(version.Minor, "11") || strings.HasPrefix(version.Minor, "10")) {
```

Comment:
I don't love this hardcoded version stuff with no comment about why it's special-cased.  In the future I suggest a small function named like isMySpecialCase(version) or some such.